schema/connector-discovery-schema.json

```diff
@@ -0,0 +1,370 @@
+      }
+    },
+    "config": {
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+        "default": {
+          "type": "string"
+        },
+        "type": {
+          "type": "string"
+        },
+        "description": {
+          "type": "object",
+          "additionalProperties": {
+            "not": {
+              "type": "object"
+            }
+          }
+        },
+        "validators": {
+          "type": "array",
+          "items": {"$ref": "#/definitions/config_validator"}
+        }
+      },
+      "required": ["description"]
+    },
+    "config_validator": {
+      "anyOf": [
+        {
+          "type": "object",
+          "additionalProperties": false,
+          "properties": {
+            "type": {
+              "const": "required"
+            },
+            "description": {
+              "type": "object",
+              "additionalProperties": {
+                "not": {
+                  "type": "object"
+                }
+              }
+            }
+          },
+          "required": ["type"]
+        },
+        {
+          "type": "object",
+          "additionalProperties": false,
+          "properties": {
+            "type": {
+              "const": "regex"
+            },
+            "value": {
+              "type": "string"
+            },
+            "description": {
+              "type": "object",
+              "additionalProperties": {
+                "not": {
+                  "type": "object"
+                }
+              }
+            }
+          },
+          "required": ["type", "value"]
+        },
+        {
+          "type": "object",
+          "additionalProperties": false,
+          "properties": {
+            "type": {
+              "const": "max_length"
+            },
+            "value": {
+              "type": "integer"
+            },
+            "description": {
+              "type": "object",
+              "additionalProperties": {
+                "not": {
+                  "type": "object"
+                }
+              }
+            }
+          },
+          "required": ["type", "value"]
+        }
+      ]
+    },
+    "input_env_variables": {
+      "type": "string",
+      "enum": [
+        "USER_EMAIL",
+        "ENROLLMENT_IDENTIFIER",
+        "ENROLLMENT_USERNAME",
+        "ENROLLMENT_GROUP_IDENTIFIER",
+        "ENROLLMENT_DEVICE_UDID",
+        "DEVICE_IDENTIFIER",
+        "DEVICE_MODEL",
+        "DEVICE_NAME",
+        "DEVICE_SYSTEM_NAME",
+        "DEVICE_SYSTEM_VERSION",
+        "DEVICE_UI_IDIOM"
+      ]
+    },
+    "http_methods": {
+      "type": "string",
+      "enum": [
+        "DELETE",
+        "GET",
+        "HEAD",
+        "OPTIONS",
+        "PATCH",
+        "POST",
+        "PUT",
+        "TRACE"
+      ]
+    },
+    "connector_action_keys": {
+      "type": "string",
+      "enum": [
+        "DIRECT",
+        "USER_INPUT",
+        "OPEN_IN"
+      ]
```

Comment:
In the above 3 enums, I think we need just 2 of those. ("DIRECT" and "USER_INPUT")

Note - Object-action allows 2 more action keys too.  "DISMISS" and "INSTALL_APP"

Reply:
I would argue DIRECT, USER_INPUT, and OPEN_IN should all be available in connector-level-actions.  Maybe INSTALL_APP, but probably we can almost forget that one exists.

Reply:
Not strict anymore against an enum. 
If a connector author has a new type, say `SWIPE_LEFT`, which their client understands, it is fine with us.